### PR TITLE
8286940: [IR Framework] Allow IR tests to build and use Whitebox without -DSkipWhiteBoxInstall=true

### PR DESCRIPTION
**Pull request by chhagedorn**
Currently, the IR framework always tries to install the Whitebox by moving the Whitebox class file to the JTreg class path. However, when a test already builds the Whitebox and uses it as part of the test, we cannot access it on certain platforms. On Windows, for example, we'll get the following exception:
```
Caused by: java.nio.file.FileSystemException: sun\hotspot\WhiteBox.class: The process cannot access the file because it is being used by another process
```
To mitigate this problem, one can specify `-DSkipWhiteBoxInstall=true` which was already done in [JDK-8283187](https://bugs.openjdk.java.net/browse/JDK-8283187). But this is not a good solution as the user should not need to worry about the inner workings of the IR framework.

I propose to get rid of this flag by reworking the Whitebox installation process.

Thanks,
Christian

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8286940](https://bugs.openjdk.org/browse/JDK-8286940): [IR Framework] Allow IR tests to build and use Whitebox without -DSkipWhiteBoxInstall=true

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [0b0b7d37](https://git.openjdk.java.net/jdk/pull/8879/files/0b0b7d371233e4117ae5f3b88efccd519be67528)
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [0b0b7d37](https://git.openjdk.java.net/jdk/pull/8879/files/0b0b7d371233e4117ae5f3b88efccd519be67528)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8879/head:pull/8879` \
`$ git checkout pull/8879`

Update a local copy of the PR: \
`$ git checkout pull/8879` \
`$ git pull https://git.openjdk.java.net/jdk pull/8879/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8879`

View PR using the GUI difftool: \
`$ git pr show -t 8879`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8879.diff">https://git.openjdk.java.net/jdk/pull/8879.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by chhagedorn** [comment]
Thanks Vladimir for your review!

File: test/hotspot/jtreg/compiler/lib/ir_framework/TestFramework.java [hunk 1]
```diff
@@ -336,6 +341,28 @@ public void start() {
         }
     }
 
+    /**
+     * Try to load the Whitebox class with a user directory custom class loader. If the user has already built the
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/compiler/lib/ir_framework/TestFramework.java:345]:
```suggestion
     * Try to load the Whitebox class from the user directory with a custom class loader. If the user has already built the
```

**Review by TobiHartmann — APPROVED** [review]
Looks good.

**Comment by chhagedorn** [comment]
Thanks Tobias for your review!

**Pull request closed without merging**